Kubernetes/containerd/start.ps1

```diff
@@ -0,0 +1,493 @@
+New-Item -ItemType Directory -Path $cniConfigDir -Force > $null
+New-Item -ItemType Directory -Path $containerdPath -Force > $null
+New-Item -ItemType Directory -Path $flanneldPath -Force > $null
+New-Item -ItemType Directory -Path $flanneldConfPath -Force > $null
+New-Item -ItemType Directory -Path $lcowPath -Force > $null
+
+# Setup functions
+
+Function CreateVMSwitch() {
+    # make the switch with internet access a hyper-v switch, if it's not one already
+    $interfaceIndex = (Get-WmiObject win32_networkadapterconfiguration | Where-Object {$null -ne $_.defaultipgateway}).InterfaceIndex
+    $netAdapter = Get-NetAdapter -InterfaceIndex $interfaceIndex
+    If(-not ($netAdapter.DriverDescription -Like "*Hyper-V*")) {
+        Write-Output "Creating VM switch"
+        New-VMSwitch -SwitchName $netAdapter.Name -AllowManagementOS $true -NetAdapterName $netAdapter.Name *>&1
+    }
+}
+
+# From https://social.technet.microsoft.com/Forums/en-US/5aa53fef-5229-4313-a035-8b3a38ab93f5/unzip-gz-files-using-powershell?forum=winserverpowershell
+Function Expand-GZip($infile, $outfile = ($infile -replace '\.gz$','')) {
+    $input = New-Object System.IO.FileStream $inFile, ([IO.FileMode]::Open), ([IO.FileAccess]::Read), ([IO.FileShare]::Read)
+    $output = New-Object System.IO.FileStream $outFile, ([IO.FileMode]::Create), ([IO.FileAccess]::Write), ([IO.FileShare]::None)
+    $gzipStream = New-Object System.IO.Compression.GzipStream $input, ([IO.Compression.CompressionMode]::Decompress)
+
+    $buffer = New-Object byte[](1024)
+    while($true){
+        $read = $gzipstream.Read($buffer, 0, 1024)
+        if ($read -le 0){break}
+        $output.Write($buffer, 0, $read)
+    }
+
+    $gzipStream.Close()
+    $output.Close()
+    $input.Close()
+}
+
+Function DownloadAndExtractTarGz($url, $dstPath) {
+    $tmpTarGz = New-TemporaryFile | Rename-Item -NewName { $_ -replace 'tmp$', 'tar.gz' } -PassThru
+    $tmpTar = New-TemporaryFile | Rename-Item -NewName { $_ -replace 'tmp$', 'tar' } -PassThru
+
+    Invoke-WebRequest $url -o $tmpTarGz.FullName
+    Expand-GZip $tmpTarGz.FullName $tmpTar.FullName
+    Expand-7Zip $tmpTar.FullName $dstPath
+    Remove-Item $tmpTarGz.FullName,$tmpTar.FullName
+}
+
+Function DownloadAndExtractZip($url, $dstPath) {
+    $tmpZip = New-TemporaryFile | Rename-Item -NewName { $_ -replace 'tmp$', 'zip' } -PassThru
+    Invoke-WebRequest $url -o $tmpZip.FullName
+    Expand-Archive $tmpZip.FullName $dstPath
+    Remove-Item $tmpZip.FullName
+}
+
+Function TestAndDownloadFile($url, $destPath, $name) {
+    $dest = Join-Path $destPath $name
+    if(-not (Test-Path $dest)) {
+        Write-Output "Downloading $name"
+        Invoke-WebRequest $url -o $dest
+    }
+}
+
+Function DownloadAllFiles() {
+    # download k8s binaries
+    TestAndDownloadFile https://storage.googleapis.com/kubernetes-release/release/v1.13.0/bin/windows/amd64/kubeadm.exe $kubernetesPath kubeadm.exe

```

Comment:
Can you take a kubernetes release as a param & use it here, rather than hard coding. Have some default value to latest.

Reply:
@daschott Can you also look at this code? We should get this in by Friday